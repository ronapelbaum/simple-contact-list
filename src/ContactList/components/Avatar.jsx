import React from 'react';
import {ReactComponent as CitizenIcon} from '../../common/images/citizen.svg';
import {ReactComponent as ProfessionalIcon} from '../../common/images/professional.svg';

const Avatar = ({ type, className}) => {
  switch (type.trim().toLowerCase()) {
    case 'citizen':
      return <CitizenIcon className={className} />
    case 'professional':
      return <ProfessionalIcon className={className} />
    default:
      return null;
  }
};

export default Avatar;
