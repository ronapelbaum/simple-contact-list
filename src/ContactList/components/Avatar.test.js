import React from "react";
import { shallow } from "enzyme";
import Avatar from "./Avatar";

describe("Avatar.jsx suite", () => {
  it("renders correctly", () => {
    expect(shallow(<Avatar />).debug()).toMatchSnapshot();
  });
  it("renders correctly Avatar of citizen", () => {
    expect(shallow(<Avatar type="citizen" />).debug()).toMatchSnapshot();
  });
  it("renders correctly Avatar of professional", () => {
    expect(
      shallow(<Avatar type=" Professional  " className="a-clazz" />).debug()
    ).toMatchSnapshot();
  });
});
