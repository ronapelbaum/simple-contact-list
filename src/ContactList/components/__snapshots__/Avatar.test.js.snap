// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar.jsx suite renders correctly 1`] = `""`;

exports[`Avatar.jsx suite renders correctly Avatar of citizen 1`] = `"<ForwardRef(SvgCitizen) className={[undefined]} />"`;

exports[`Avatar.jsx suite renders correctly Avatar of professional 1`] = `"<ForwardRef(SvgProfessional) className=\\"a-clazz\\" />"`;
